sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/AggregateEstimation.scala

```diff
@@ -42,8 +42,8 @@ object AggregateEstimation {
         (res, expr) => {
           val columnStat = childStats.attributeStats(expr.asInstanceOf[Attribute])
           val distinctCount = columnStat.distinctCount.get
-          val distinctValue: BigInt = if (distinctCount == 0 && columnStat.nullCount.get > 0) {
-            1
+          val distinctValue: BigInt = if (columnStat.nullCount.get > 0) {
+            distinctCount + 1
```

Comment:
Hm, actually, do we need to count `null` as distinct value? It's not counted as a distinct value in SQL (`F.countDistinct` or `count(DISTINCT col)`) and Pandas (`unique()` by default) at least.

Reply:
Looking into the current impl, seems we ignore `null` as distinct values:

https://github.com/apache/spark/blob/239082d9667a4fa4198bd9524d63c739df147e0e/sql/core/src/main/scala/org/apache/spark/sql/execution/command/CommandUtils.scala#L270
https://github.com/apache/spark/blob/b1857a4d7dfe17663f8adccd7825d890ae70d2a1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/EstimationUtils.scala#L50-L51